compiler/rustc_query_system/src/query/plumbing.rs

```diff
@@ -346,6 +251,67 @@ where
     }
 }
 
+#[cold]
+#[inline(never)]
+#[cfg(not(parallel_compiler))]
+fn cycle_error<Q, Qcx>(
+    query: Q,
+    qcx: Qcx,
+    try_execute: QueryJobId,
+    span: Span,
+) -> (Q::Value, Option<DepNodeIndex>)
+where
+    Q: QueryConfig<Qcx>,
+    Qcx: QueryContext,
+{
+    // If we are single-threaded we know that we have cycle error,
+    // so we just return the error.
```

Comment:
Could you move the comment at the caller site?

Reply:
Done.